BridgeSDK/BridgeSDK.m

```diff
@@ -61,6 +65,20 @@ + (void)setupWithStudy:(NSString *)study useCache:(BOOL)useCache environment:(SB
     gSBBAppStudy = study;
     gSBBDefaultEnvironment = environment;
     gSBBUseCache = useCache;
+    gSBBCacheDaysAhead = SBBDefaultCacheDaysAhead;
+    gSBBCacheDaysBehind = SBBDefaultCacheDaysBehind;
+    
+    // make sure the Bridge network manager is set up as the delegate for the background session
+    [SBBComponent(SBBBridgeNetworkManager) restoreBackgroundSession:kBackgroundSessionIdentifier completionHandler:nil];
+}
+
++ (void)setupWithStudy:(NSString *)study cacheDaysAhead:(NSInteger)cacheDaysAhead cacheDaysBehind:(NSInteger)cacheDaysBehind environment:(SBBEnvironment)environment
+{
+    gSBBAppStudy = study;
+    gSBBDefaultEnvironment = environment;
+    gSBBUseCache = cacheDaysAhead > 0 || cacheDaysBehind > 0;
+    gSBBCacheDaysAhead = MIN(SBBMaxSupportedCacheDays, cacheDaysAhead);
```

Comment:
In the words of the Rolling Stones, "You can't always get what you want".  :)

Alx was concerned that allowing for a longer "time in advance" could have a serious performance hit for applications that have a lot of scheduled activities. Because of this, he implemented logic as follows:

1. Get up to 4 days ahead of schedules, based on the value of the "daysAhead" parameter.
2. If the number of returned activities for a given schedule is less than "minimumPerSchedule" then return a maximum of that number of activities for each schedule.

Both of these numbers are capped at 4 and the server will return an error if you send a value greater than that.  However, in order to make his server logic work with the existing caching architecture that we have in place (which uses a simpler design of a time-banded range of days ahead and days behind), I had to fudge it.

While we could re-architect the caching to use the same daysAhead/daysBehind/minimumPerSchedule architecture, that was more effort AND I think it's confusing.  So instead I hide these implementation details.

The purpose of `SBBMaxSupportedCacheDays` is really intended to limit the "days behind" caching range since the current architecture (of grabbing the entire cache of schedules and then writing it back to Core Data rather than using a more complicated data sync algorithm) would cause performance issues if the number of records gets too big. I felt that solving that problem was a different issue that could be prioritized in JIRA with it's own ticket.

This pull request is intended to address the more immediately pressing issue of "Smart4SURE needs to be able to get 10 days ahead and the server only supports 4 days ahead or a maximum of 4 activities per schedule".